sdk/src/types/block/payload/signed_transaction/transaction.rs

```diff
@@ -439,6 +444,30 @@ fn verify_outputs(outputs: &[Output], visitor: &ProtocolParameters) -> Result<()
     Ok(())
 }
 
+fn verify_transaction_packable(transaction: &Transaction, _: &ProtocolParameters) -> Result<(), PayloadError> {
+    verify_transaction(transaction)
+}
+
+fn verify_transaction(transaction: &Transaction) -> Result<(), PayloadError> {
+    let has_commitment_input = transaction.context_inputs().commitment().is_some();
+
+    for output in transaction.outputs.iter() {
+        if output.features().is_some_and(|f| f.staking().is_some()) && !has_commitment_input {
+            return Err(PayloadError::MissingCommitmentInputForStakingFeature);
+        }
+
+        if output.features().is_some_and(|f| f.block_issuer().is_some()) && !has_commitment_input {
+            return Err(PayloadError::MissingCommitmentInputForBlockIssuerFeature);
+        }
+
+        if output.is_delegation() && !has_commitment_input {
+            return Err(PayloadError::MissingCommitmentInputForDelegationOutput);
+        }
+    }
```

Comment:
Couldn't you running through the loop if `has_commitment_input` is `true`?

Reply:
I can probably reverse the condition yeah, will give it a try

Reply:
Done, thanks